In [631]:
#Import pandas 
import pandas as pd

In [ ]:
#Load the dataset 
df = pd.read_csv(r"C:\Users\marya\OneDrive\Desktop\fintech\finpay-data-analytics-roadmap\datasets\customers_dirty.csv")
df

,customer_id,first_name,last_name,gender,age,city,country,signup_date,employment_status,annual_income,credit_score,risk_level
0,C001,Sara,Khan,Female,35,Jeddah,Saudi Arabia,2024-03-12,Student,66868,846.0,High
1,C002,Maryam,Iqbal,Female,22,Riyadh,Saudi Arabia,2024-02-17,Salaried,100990,758.0,High
2,C003,Ali,Qureshi,Male,54,Khobar,Saudi Arabia,2024-04-22,Self-Employed,112926,503.0,Low
3,C004,Ahmed,Malik,Female,29,Jeddah,Saudi Arabia,2025-05-05,Self-Employed,66793,547.0,Medium
4,C005,Sara,Malik,Female,58,Dammam,Saudi Arabia,2025-02-17,Salaried,NaN,774.0,Low
...,...,...,...,...,...,...,...,...,...,...,...,...
97,C098,Lina,Shaikh,Female,55,Riyadh,Saudi Arabia,2024-08-20,Salaried,122448,629.0,Medium
98,C099,Sara,Hussain,Male,54,Khobar,Saudi Arabia,2024-07-30,Salaried,89173,765.0,Medium
99,C100,Zaid,Farooq,Female,23,Jeddah,Saudi Arabia,2024-05-16,Student,74344,647.0,Medium
100,C010,Ali,Ansari,Male,40,Khobar,Saudi Arabia,2024-05-17,Salaried,95307,790.0,High


In [633]:
#Display the first 10 rows
df.head()

,customer_id,first_name,last_name,gender,age,city,country,signup_date,employment_status,annual_income,credit_score,risk_level
0,C001,Sara,Khan,Female,35,Jeddah,Saudi Arabia,2024-03-12,Student,66868,846.0,High
1,C002,Maryam,Iqbal,Female,22,Riyadh,Saudi Arabia,2024-02-17,Salaried,100990,758.0,High
2,C003,Ali,Qureshi,Male,54,Khobar,Saudi Arabia,2024-04-22,Self-Employed,112926,503.0,Low
3,C004,Ahmed,Malik,Female,29,Jeddah,Saudi Arabia,2025-05-05,Self-Employed,66793,547.0,Medium
4,C005,Sara,Malik,Female,58,Dammam,Saudi Arabia,2025-02-17,Salaried,NaN,774.0,Low


In [634]:
#Display the last 10 rows 
df.tail()

,customer_id,first_name,last_name,gender,age,city,country,signup_date,employment_status,annual_income,credit_score,risk_level
97,C098,Lina,Shaikh,Female,55,Riyadh,Saudi Arabia,2024-08-20,Salaried,122448,629.0,Medium
98,C099,Sara,Hussain,Male,54,Khobar,Saudi Arabia,2024-07-30,Salaried,89173,765.0,Medium
99,C100,Zaid,Farooq,Female,23,Jeddah,Saudi Arabia,2024-05-16,Student,74344,647.0,Medium
100,C010,Ali,Ansari,Male,40,Khobar,Saudi Arabia,2024-05-17,Salaried,95307,790.0,High
101,C041,Different,Qureshi,Male,27,Riyadh,Saudi Arabia,2025-04-28,Student,79073,779.0,Low


In [635]:
#How many rows and columns?
df.shape

(102, 12)

In [636]:
#Display column names
df.columns

Index(['customer_id', 'first_name', 'last_name', 'gender', 'age', 'city',
       'country', 'signup_date', 'employment_status', 'annual_income',
       'credit_score', 'risk_level'],
      dtype='str')

In [637]:
#Display data types
df.dtypes

customer_id              str
first_name               str
last_name                str
gender                   str
age                    int64
city                     str
country                  str
signup_date              str
employment_status        str
annual_income            str
credit_score         float64
risk_level               str
dtype: object

In [638]:
#Display dataset information
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 102 entries, 0 to 101
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   customer_id        102 non-null    str    
 1   first_name         101 non-null    str    
 2   last_name          102 non-null    str    
 3   gender             102 non-null    str    
 4   age                102 non-null    int64  
 5   city               102 non-null    str    
 6   country            102 non-null    str    
 7   signup_date        102 non-null    str    
 8   employment_status  101 non-null    str    
 9   annual_income      101 non-null    str    
 10  credit_score       101 non-null    float64
 11  risk_level         102 non-null    str    
dtypes: float64(1), int64(1), str(10)
memory usage: 9.7 KB


In [639]:
#Summary Statistics for numeric columns 
df.describe()

,age,credit_score
count,102.000000,101.000000
mean,40.960784,691.237624
std,16.504378,115.762615
min,-5.000000,150.000000
25%,33.000000,618.000000
50%,40.000000,696.000000
75%,50.750000,782.000000
max,150.000000,950.000000


In [641]:
#Check missing values
df.isnull().sum()

customer_id          0
first_name           1
last_name            0
gender               0
age                  0
city                 0
country              0
signup_date          0
employment_status    1
annual_income        1
credit_score         1
risk_level           0
dtype: int64

In [642]:
#Count total missing values 
df.isnull().sum().sum()

np.int64(4)

In [643]:
#Count duplicate rows 
df.duplicated().sum()

np.int64(1)

In [644]:
#Displays all duplicate rows in the dataframe
df[df.duplicated()]

,customer_id,first_name,last_name,gender,age,city,country,signup_date,employment_status,annual_income,credit_score,risk_level
100,C010,Ali,Ansari,Male,40,Khobar,Saudi Arabia,2024-05-17,Salaried,95307,790.0,High


In [645]:
#check unique values
df.nunique()

customer_id          100
first_name            11
last_name             10
gender                 5
age                   39
city                   7
country                2
signup_date           91
employment_status      5
annual_income         99
credit_score          89
risk_level             3
dtype: int64

Frequency Analysis

In [646]:
df["city"].value_counts()

city
Riyadh      24
Khobar      23
Jeddah      19
Dammam      18
Mecca       16
 riyadh      1
RIYADH       1
Name: count, dtype: int64

In [647]:
df["employment_status"].value_counts()

employment_status
Student          35
Self-Employed    34
Salaried         30
Unknown           1
-                 1
Name: count, dtype: int64

In [648]:
df["gender"].value_counts()

gender
Female    51
Male      48
 male      1
MALE       1
F          1
Name: count, dtype: int64

In [649]:
df["risk_level"].value_counts()

risk_level
Low       40
High      31
Medium    31
Name: count, dtype: int64

Numerical Exploration

In [650]:
df["age"].min()

np.int64(-5)

In [651]:
df["age"].max()

np.int64(150)

In [652]:
df["age"].mean()

np.float64(40.96078431372549)

In [663]:
df["annual_income"] = pd.to_numeric(df["annual_income"],errors="coerce")

In [667]:
df["annual_income"].mean()

np.float64(107705.67)

In [664]:
df["credit_score"].mean()

np.float64(691.2376237623762)

Business Questions:

How many customers are there?

102 customers


What information do we collect?

The datasets collect information about each customer's:
Customer ID

First Name 

Last Name

Gender

Age

City

Country

Signup Date 

Employment Status

Annual Income 

Credit Score 

Risk Level

Which columns are categorical?

customer_id, first_name, last_name, gender, city, country, employment_status, risk_level

Which columns are numerical?

age, annual_income, credit_score


Which columns might need cleaning?

first_name, employment_status, annual_income, credit_risk, signup-date, city, gender,customer_id

What could each department (Finance, Marketing, Risk, Operations) use from this dataset?

Finance Team - Understand customer financial strength and potential profitability

Marketing Team - Segment customers and plan targeted campaigns

Risk Team - Evaluate whether customers are likely to repay BNPL installments

Operations Team - Maintain accurate customer records and support daily operations 